controllers/user.js

```diff
@@ -0,0 +1,106 @@
+import dotenv from 'dotenv';
+import jwt from 'jsonwebtoken';
+import { User } from '../models';
+// import sendMail from '../helpers/emails';
+
+dotenv.config();
+const { JWT_SECRET } = process.env;
+const generateToken = id => jwt.sign(
+  { id },
+  JWT_SECRET,
+  { expiresIn: '24h' },
+);
+
+
+/**
+ * @class UserController
+ *  @override
+ * @export
+ *
+ */
+export default class UserController {
+  /**
+     * @description - Creates a new user
+     * @static
+     *
+     * @param {object} req - HTTP Request
+     * @param {object} res - HTTP Response
+     *
+     * @memberof UserController
+     *
+     * @returns {object} Class instance
+     */
+  static registerUser(req, res) {
+    const { body } = req;
+    User.create(body)
+      .then((newUser) => {
+        const {
+          dataValues: {
+            id,
+          },
+        } = newUser;
+        const token = generateToken(id);
+        return res.status(201)
+          .json({
+            success: true,
+            message: 'You have signed up successfully.',
+            token,
+          });
+      })
+      .catch((error) => {
+        const errors = [];
+        if (error.errors[0].path === 'username') {
+          errors.push(error.errors[0].message);
+        }
+        if (error.errors[0].path === 'email') {
+          errors.push(error.errors[0].message);
+        }
+        return res.status(409)
+          .json({
+            success: false,
+            errors,
+          });
+      });
+  }
+
+  /**
+     * @description - Verifies a user's account
+     * @static
+     *
+     * @param {object} req - HTTP Request
+     * @param {object} res - HTTP Response
+     *
+     * @memberof UserController
+     *
+     * @returns {object} Class instance
+     */
+  static verifyAccount(req, res) {
+    const { params: { verificationId } } = req;
+    User.update({
+      verified: true,
+    }, {
+      where: {
+        verificationId,
+        verified: false,
+      },
+    })
+      .then((updatedUser) => {
+        if (updatedUser[0] === 0) {
+          return res.status(422)
+            .json({
+              success: false,
+              message: 'Error: This user is either verified or not found.',
```

Comment:
@leksyib Don't you think the user's verification ID should be deleted or set to NULL after they're verified? This way, the error will always be that the user doesn't exist

Reply:
Perhaps another db hook to achieve that?

Reply:
Alright, I'll implement that.